packages/form-state-manager/src/utils/use-subscription.ts

```diff
@@ -157,33 +122,49 @@ const useSubscription = ({
     if (validate) {
       const error = fieldLevelValidator(validate, sanitizedValue, formOptions().values, formOptions);
       if (isPromise(error)) {
-        setState((prevState) => ({ ...prevState, meta: { ...prevState.meta, validating: true } }));
+        setState((prevState: FieldState) => ({ ...prevState, meta: { ...prevState.meta, validating: true } }));
         const asyncError = error as Promise<string | undefined>;
         asyncError.then(() => handleError(true)).catch((error) => handleError(false, error));
       } else {
         const syncError = error as string | undefined;
         if (error) {
           handleError(false, syncError);
-        } else if (state.meta.valid === false) {
+        } else if (formOptions().getFieldState(name)?.meta.valid === false) {
           handleError(true);
         }
       }
     }
 
-    setState((prevState) => ({ ...prevState, value: getFieldValue(name) }));
+    setState((prevState: FieldState) => ({ ...prevState, value: getFieldValue(name) }));
   };
 
-  const valueToReturn = state.value;
+  const valueToReturn = formOptions().getFieldValue(name);
+
+  const { current: initValue } = useRef(initialValue || get(initialValues, name));
 
   useEffect(() => {
+    const values = formOptions().values;
+    const firstinitialization = !formOptions().initializedFields.includes(name);
+    let value = get(values, name);
+
+    if (firstinitialization || shouldExecute(formOptions().initializeOnMount, initializeOnMount)) {
+      value = initValue;
+    }
+
+    const state = {
+      value,
+      name,
+      meta: initialMeta(initValue),
+      internalId: generateId()
+    };
```

Comment:
I think this initialization code should be handled in `managerApi` -- schematic

i.e.:

```jsx
// manager-api

....
function registerField() {
    addToFieldListeners(internalId, render);
    if(initializeOnMount) { change(value) }; // compute initialValue/value

    if(!state[field]) {
      state[field] = createField(); // << initialMeta()
    }

   return state[field];
}
...

```